storage/innobase/trx/trx0purge.cc

```diff
@@ -1409,6 +1381,12 @@ static void trx_purge_wait_for_workers_to_complete()
 TRANSACTIONAL_INLINE
 void purge_sys_t::clone_end_view(const purge_sys_t::iterator &head)
 {
+  /* Release the undo pages. */
+  for (auto p : pages)
```

Comment:
There is only one caller of clone_end_view(). And it's said that it just updates end_view in the documentation. It does not look like a callback to clean everything after purging. I propose to move the code of pages cleaning to clone_end_view() caller.

Reply:
The reason why I added the cleanup here is that the single caller is not a member function of `purge_sys_t`. We could add another single-caller function to do the cleanup, or maybe a better solution would be to rename the `clone_end_view()` to `batch_cleanup()` or something. The name `purge_sys_t::clone_end_view()` was introduced in ab0190101b0587e0e03b2d75a967050b9a85fd1b in 10.6.

Reply:
9ac62e81be6564596c3f92a47166d42c9d89eca0 renames `clone_end_view()` to `batch_cleanup()`.